pkg/s3wrapper/filesystem.go

```diff
@@ -56,6 +60,7 @@ func (f *FSClient) Upload(ctx context.Context, data []byte, objectName string) e
 		log.Error(err)
 		return err
 	}
+	f.reportFilesystemUsageMetrics()
```

Comment:
Adding this manually in every method is error-prone.  Is there a better way to do it?  If this was Python we'd probably use a decorator.

Reply:
@avishayt 
I added a decorator to wrap the FSClient and be responsible for collecting metrics when needed.
Haven't used any reflection voodoo which is less encouraged in golang.
Pls let me know what is your opinion about it